llvm/lib/Target/X86/lab3/korablev_nikita/X86KorablevMICounterPass.cpp

```diff
@@ -0,0 +1,50 @@
+#include "X86.h"
+#include "X86InstrInfo.h"
+#include "X86Subtarget.h"
+#include "llvm/CodeGen/MachineFunctionPass.h"
+#include "llvm/CodeGen/MachineInstrBuilder.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+
+using namespace llvm;
+
+namespace {
+class X86KorablevMICounterPass : public MachineFunctionPass {
+public:
+  static char ID;
+  X86KorablevMICounterPass() : MachineFunctionPass(ID) {}
+
+  bool runOnMachineFunction(MachineFunction &MF) override {
+    const TargetInstrInfo *TII = MF.getSubtarget().getInstrInfo();
+    DebugLoc DL = MF.front().begin()->getDebugLoc();
+
+    Module &module = *MF.getFunction().getParent();
+    GlobalVariable *var = module.getGlobalVariable("ic");
+
+    if (!var) {
+      var = new GlobalVariable(module,
+                               IntegerType::get(module.getContext(), 64), false,
+                               GlobalValue::ExternalLinkage, nullptr, "ic");
+    }
+
+    for (auto &BB : MF) {
+      size_t count = 0;
+      for (auto MI = BB.begin(); MI != BB.end(); ++MI)
+        ++count;
```

Comment:
Can't we take BB length in a faster and simpler way?

Reply:
Yes, you're right